File: platform-core-service/platform-core-service.Common/Models/DTOs/EntityDTO/Post/SelectPostDTO.cs
using platform_core_service.Common.Entities.BaseEntity;
using platform_core_service.Common.Utils.Enums;

namespace platform_core_service.Common.Models.DTOs.EntityDTO.Post
{
    public class SelectPostDTO : IBaseKey<string>
    {
        public string Id { get; set; } = null!;

        public string Title { get; set; } = null!;

        public string Content { get; set; } = null!;

        public string Slug { get; set; } = null!;

        public PostType PostType { get; set; }

        public ModerationStatus ModerationStatus { get; set; }

        public string? ModerationReason { get; set; }

        public string AuthorId { get; set; } = null!;

        public SelectPostAuthorDTO? Author { get; set; }

        public int UpvoteCount { get; set; } = 0;
        public int DownvoteCount { get; set; } = 0;
        public int CommentCount { get; set; } = 0;

        public List<string> TagNames { get; set; } = new List<string>();

        public DateTimeOffset DateCreated { get; set; }

        public DateTimeOffset? DateModified { get; set; }
        public bool? CurrentUserVote { get; set; }

        public string? CommunityId { get; set; }
        public SelectPostCommunityDTO? Community { get; set; }
    }

    public class SelectPostAuthorDTO
    {
        public string Id { get; set; } = null!;
        public string FullName { get; set; } = null!;
        public string? AvatarUrl { get; set; }
        public List<string> TechStacks { get; set; } = new List<string>();
    }

    public class SelectPostCommunityDTO
    {
        public string Id { get; set; } = null!;
        public string Name { get; set; } = null!;
        public string Slug { get; set; } = null!;
        public string? CommunityCoverPhotoUrl { get; set; }
    }
}

File: frontend/src/types/post/select-post-dto.ts
import { PostType } from "./create-post-dto";
import { ModerationStatus } from "./moderation-status";
import { CommunityApprovalStatusValue } from "../enums/community-approval-status";

export interface SelectPostAuthorDTO {
    id: string;
    fullName: string;
    avatarUrl?: string;
    backgroundUrl?: string;
    bio: string;
    reputationPoints: number;
    techStacks: string[];
    isPrivate: boolean;
}

export interface SelectPostCommunityDTO {
    id: string;
    name: string;
    slug: string;
    communityCoverPhotoUrl?: string;
    description?: string | null;
    memberCount?: number;
    isPrivate?: boolean;
}

export enum SharedContentType {
    Post = 0,
    QAPost = 1,
}

export interface SelectSharedPostDTO {
    id: string;
    title: string;
    contentPreview: string;
    communityId?: string | null;
    community?: SelectPostCommunityDTO | null;
    author?: SelectPostAuthorDTO | null;
    contentType: SharedContentType;
    dateCreated: string;
}

export interface SelectPostDTO {
    id: string;
    title: string;
    content: string;
    slug: string;
    postType: PostType;
    moderationStatus: ModerationStatus;
    moderationReason?: string | null;
    communityApprovalStatus?: CommunityApprovalStatusValue | null;
    communityApprovalReason?: string | null;
    authorId: string;
    author?: SelectPostAuthorDTO;
    upvoteCount: number;
    downvoteCount: number;
    commentCount: number;
    tagNames: string[];
    dateCreated: string;
    dateModified: string;
    currentUserVote?: boolean | null; // null = no vote, true = upvoted, false = downvoted
    communityId?: string;
    community?: SelectPostCommunityDTO;
    sharedPostId?: string | null;
    sharedPost?: SelectSharedPostDTO | null;
    isSaved: boolean;
    savedBookMarkedItemId?: string;
}
